modules/filestore/amazonS3/amaxonS3.go

```diff
@@ -0,0 +1,43 @@
+package amazonS3
+
+import (
+	"fmt"
+	"os"
+
+	"github.com/aws/aws-sdk-go/aws"
+	"github.com/aws/aws-sdk-go/aws/session"
+	"github.com/spaceuptech/space-cloud/utils"
+)
+
+type AmazonS3 struct {
+	tempFileName string
+	session      *session.Session
+}
+
+func Init(region string) (*AmazonS3, error) {
+	sess, err := session.NewSession(&aws.Config{
+		Region: aws.String(region)},
+	)
+	if err != nil {
+		fmt.Println("AmazonS3 Couldn't Establish Connection ", err)
+		return nil, err
+	}
+	_, err = sess.Config.Credentials.Get()
+
+	if err != nil {
+		fmt.Println("AmazonS3 Credentials Not Found ", err)
+		return nil, err
+	}
+	return &AmazonS3{"", sess}, nil
+}
+
+// GetStoreType returns the file store type
+func (a *AmazonS3) GetStoreType() utils.FileStoreType {
+	return utils.AmazonS3
+}
+
+// Close gracefully closed the local filestore module
+func (a *AmazonS3) Close() error {
```

Comment:
ohh ,i forgot to close the session. ill do that
from where to delete the temp file

Reply:
Yes. Close the session. I've mentioned where you can delete the temp file

Reply:
there is close method for session.
from internet
Yup. No need to close or delete the client. Except where explicitly indicated otherwise, each call is a distinct API request to an endpoint for that service with no long-lived connection maintained. 
from aws
You can use the AWS Systems Manager console or the AWS CLI to terminate a session that you started to connect to an instance in your account. By default, if there is no user activity after 20 minutes, a session is terminated. After a session is terminated, it can't be resumed. 
if that's the case should the session be in struct or should i request for a new session every time the function is called ?

Reply:
there is no close method for closing the session
from internet
Yup. No need to close or delete the client. Except where explicitly indicated otherwise, each call is a distinct API request to an endpoint for that service with no long-lived connection maintained. 
from aws
You can use the AWS Systems Manager console or the AWS CLI to terminate a session that you started to connect to an instance in your account. By default, if there is no user activity after 20 minutes, a session is terminated. After a session is terminated, it can't be resumed. 
in that case should i remove the session from the struct and request for a new session every time the fuction is called ?

Reply:
Can we check if the session has expired. In that case, we can check in each function call whether the session has expired. If it has, we can create a new session right there.

Otherwise we should create a new session in every function call.

Reply:
default duration is ablout 1 hour
not able check if the session has expired as it is an http call underneath 
what to do?

Reply:
Lets create a new session on every request